keygen/src/keygen.rs

```diff
@@ -310,78 +374,167 @@ fn main() -> Result<(), Box<dyn error::Error>> {
             output_keypair(&keypair, &outfile, "recovered")?;
         }
         ("grind", Some(matches)) => {
+            struct Match {
+                starts: String,
+                ends: String,
+                count: u64,
```

Comment:
I think we can turn `count` into an `AtomicU64` and delete `found` -- when a new match is found, decrement `count` and exit when it hits zero.

Reply:
updated